File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
         int profit = 0;
         int buy = 10000000,ans=0,maxi=0,flag=0;
         for(int i=0; i<prices.length; i++) {
            if(buy > prices[i]) {
                buy = prices[i];
            }
            System.out.println(buy);

            profit=prices[i] - buy;
            if(maxi < profit) {
                maxi = profit;
            }
         }
        return maxi;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int buy = INT_MAX;
        int profit = 0;
        int maxi = 0;
       for(int i=0; i<prices.size(); i++) {
        buy = min(buy,prices[i]);
        profit = prices[i]-buy;
        maxi = max(profit,maxi);
       } 
       return maxi;
    }
};